platform/core/src/services/UIDialogService/index.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * A UI Element
+ *
+ * @typedef {ReactElement|HTMLElement} DialogContent
+ */
+
+/**
+ * UI Dialog
+ *
+ * @typedef {Object} DialogProps
+ * @property {string} id -
+ */
+
+const uiDialogServicePublicAPI = {
+  name: 'UIDialogService',
+  dismiss,
+  dismissAll,
+  create,
+  setServiceImplementation,
+};
+
+const uiDialogServiceImplementation = {
+  _dismiss: () => console.warn('dismiss() NOT IMPLEMENTED'),
+  _dismissAll: () => console.warn('dismissAll() NOT IMPLEMENTED'),
+  _create: () => console.warn('create() NOT IMPLEMENTED'),
+};
+
+function createUIDialogService() {
+  return uiDialogServicePublicAPI;
+}
+
+/**
+ * Show a new UI dialog;
+ *
+ * @param {DialogContent} component React component
+ * @param {DialogProps} props { id, content, ...props }
+ */
+function create({ id, content, ...props }) {
```

Comment:
in `UIModalService`, we defined the props here. I think being explicit makes for a better contract, but I understand the want of something that's more flexible. I think we want extension authors to program to these contracts, and not toward any one implementation. By not defining props, we're saying, look to the underlying implementation to figure out what should be passed.

Reply:
Done, thx.